### DISCUSSION TITLE
Cannot approve Copilot suggestions with Tab in IntelliJ IDEA

### DISCUSSION THREAD

**Discussion by book000** [community/community#7199] [Copilot Conversations] [answered]
I use GitHub Copilot with IntelliJ IDEA 2021.2.3 (Ultimate Edition). Thank you for creating a great plugin.

I typed `long calculateDaysBetweenDates` and pressed Tab, hoping to adopt the suggestion that Copilot displayed.
![20211103-215425-idea64-mM8F1rLAxE](https://user-images.githubusercontent.com/8929706/140063642-d1078e34-aa70-4332-92a1-383e395f89db.png)

However, pressing Tab only adds a space for indentation after it, and does not insert the suggested code.
![20211103-215438-idea64-OB7JO3egRz](https://user-images.githubusercontent.com/8929706/140066393-55ee9c08-865f-4bac-b776-7554bb4d4048.png)
![image](https://user-images.githubusercontent.com/8929706/140067895-acf6798c-2fe2-4a52-916d-75ed3d5804e4.png)

Do you know of a way to solve this? I look forward to your reply.

---

The following is a description of my environment:

- IntelliJ IDEA 2021.2.3 (Ultimate Edition) `#IU-212.5457.46` (Build at `October 12, 2021`)
- Runtime version: `11.0.12+7-b1504.40 amd64`
- VM: `OpenJDK 64-Bit Server VM by JetBrains s.r.o.`
- OS: `Windows 10 10.0`
- Non-Bundled Plugins: `com.intellij.ja (212.324), com.samdark.intellij-visual-studio-code-dark-plus (2.5), com.wakatime.intellij.plugin (13.1.3), org.turbanov.run.configuration.as.action (1.4.5), com.github.lppedd.idea-conventional-commit (0.19.0), com.tabnine.TabNine (0.5.2), com.almightyalpaca.intellij.plugins.discord (1.7.1), zielu.gittoolbox (212.7.15), com.demonwav.minecraft-dev (2021.2-1.5.15), com.github.copilot (1.0.2), org.jetbrains.kotlin (212-1.5.31-release-546-IJ4638.7), cn.yiiguxing.plugin.translate (3.2.0.1)`
- Tab Keymap: ![image](https://user-images.githubusercontent.com/8929706/140068012-eeebd5af-8b73-49d2-8a37-4814b7103bad.png)

**Comment by jansorg** [comment, marked as answer]
Thanks for reporting! `Tab` is the right shortcut to use. The next update will allow to use your own shortcut.
Were you editing a Java file?

I wasn't able to reproduce this with the same set of custom plugins (except the Japanese language pack).
Is it working for you with the language pack disabled? There are currently a few known issues when a language pack is installed, but this shouldn't affect Tab.

**Reply by book000** [reply]
Thank you for your reply.

> Were you editing a Java file?

Yes. Editing a Java file in a Maven project.

> Is it working for you with the language pack disabled?

I disabled the Japanese language pack and tried again, but the situation did not improve.

---

I also tried installing IntelliJ IDEA in Windows SandBox and importing the settings and plugins, but I could not reproduce the problem.
I've heard that the next update will have its own shortcuts, so I'll wait for that for now.

Thank you for taking time out of your busy schedule to respond!

**Reply by jansorg** [reply]
The new update is available now. Could you try the new version? If Tab doesn't work, then try re-assigning it to something else. You can configure this at `Preferences > Keymap > Plugins > GitHub Copilot`.

**Reply by book000** [reply]
Thanks for providing the update. I have confirmed that Tab is now working and reassignment also works! 💪

**Reply by ovidiu-ionescu** [reply]
If I assign Ctrl-J for Github completion in the Keymap, it just ignores it.

**Comment by khanimranit** [comment]
GitHub copilot is  not working with Intellij 


IntelliJ IDEA 2023.2 (Ultimate Edition)
Build #IU-232.8660.185, built on July 26, 2023

**Comment by albertgo** [comment]
Github copilot is not working with PhpStorm

PhpStorm 2024.1.4
Version: 1.5.17.6356
